Explore page db implementation

dericky1 opened the pull request.

dericky1 committed:
Rename some functions for clairy

dericky1 committed:
Create test_companies endpoint and function

dericky1 committed:
Create companyToDiscount api

dericky1 committed:
Create endpoint to get a single company

dericky1 committed:
Create test_discount endpoint

dericky1 committed:
Update explore and detail page to use new db

LOTS of work still needs to be done but the barebones is working. Will need to clean up a lot of the code too.

brewswain committed:
Merge branch 'db-update__release' into explore-page-db-implementation

brewswain commented on fwb/app/(auth)/(routes)/detail/detail.tsx:
```diff
@@ -1,41 +1,35 @@
+  discountIds: any
 }) {
-  const [discounts, setDiscounts] = useState<DiscountDataDetail[]>([])
+  const [discounts, setDiscounts] = useState<any[]>([])
```
For now this can stay as any, but I found a pretty cool supabase function we can put on all our calls to state what our expected return type should be, it'll be the format of:
 ```
   let { data: groupsData, error: groupsDataError } = await supabase
      .from('test_groups')
      .select('*')
      .in('id', groupIds)
      .returns<Group[]>()

```

Where .returns< T >() will make life easier for us!

brewswain commented on fwb/app/(auth)/(routes)/detail/detail.tsx:
```diff
@@ -213,8 +213,8 @@ function DetailPageContent({ data }: { data: DetailData }) {
-            (item: DiscountDataDetail) => (
-              <DetailCard data={item} key={item.discount_amount} />
+            (item: any, index: number) => (
+              <DetailCard data={item} key={index} />
```
Definitely way better than the discount_amount being the key 😭
Once we type this properly though, it'll prob be better to have the item's id. Not important for right now.

brewswain commented on fwb/app/(auth)/(routes)/detail/page.tsx:
```diff
@@ -1,15 +1,73 @@
-} from '@/components/ui/explore/filter_context'
+import DetailPage from './detail'
+
+export async function getCompanyId(companyName: string) {
```
This is causing the build problem, remove any export functions into a separate util file, and import it from there. 
So in this case, shift out getCompanyId() & getCompanyDiscountsTable(). handleSearch can stay since we're not exporting it.

Pages don't allow for us to have multiple exports like this

brewswain reviewed (commented):
Once you fix the highlighted Build error causing code in our details/page.tsx, this'll be good to merge 👍🏽

brewswain committed:
Build fix

- Checked for places where exported functions were used, but didn't see anything.
- Will revert if we encounter any weird bugs
